protocol/0057-TRAN-transfers.md

```diff
@@ -67,6 +67,9 @@ A recurring transfers needs to contain this specific information:
 - start epoch: at the end of this epoch the first recurring transfer will be made between
 - end epoch (optional): at the end of this epoch the last recurring transfer will be made between, optional. If not specified the transfer run until cancelled (by its creator or by the network as described below).
 - factor, decimal > 0.0 (a factor used with the amount specified for the transfer).
+- transfer interval: number of epochs between transfers, i.e. when 4, funds will be transferred every 4 epochs with the first transfer occurring 4 epochs after the transaction is processed. Must be an integer strictly greater than `0` and less than `100` (this ceiling is for performance reasons and matches the limit on the `window_length` field).
+
+Note, to avoid parties intentionally creating transfers which never occur, either between parties or to fund a [reward pool](#recurring-transfers-to-reward-accounts). The funds for the next scheduled transfer should be taken straight away from the origin account, locked by the network, and distributed to the destination account at the end of the correct epoch. This way, if a user cancels a recurring transfer or withdraws all their funds, the next scheduled transfer will still take place.
```

Comment:
this is not straightforward. reward transfers do not go through unless there's eligible parties, so how is this gonna work? we first transfer the transfer amount to some temp account and then at the relevant epoch we either return it (and park the amount for the next round)? Also this also means that a transfer cannot be cancelled and is guaranteed to go at least once. I don't think this is desired. If this is spam protection maybe we need to think of an alternative.

Reply:
Less spam protection (although we did discuss how it could help with that) and more a prevention of users creating "fake" competitions. 

e.g. I create a new market, I want to attract users so I create a large reward with a first payout far in the future (say  one month). Users then trade expecting rewards but at the end of the month I cancel the transfer or the transfer fails for whatever reason. Users left without rewards.

In the case of no eligible parties I think the flow you described is fine, "return funds then park funds for the next round."  In the case a user wants to cancel a transfer because a market has no activity, they could cancel the transfer and after the next relevant payout epoch, if there are still no eligible parties they would have their funds returned.

Reply:
Anyone can make a mistake in setting up transfers as we know well, and such a mistake can be very expensive as you're guaranteeing that at least one round of payout will be made and you can't do anything to prevent it once the transfer has been included in a block.